apps/authorization/models.py

```diff
@@ -70,11 +73,217 @@ def update_grants(*args, **kwargs):
+        .distinct()
+        .count()
+    )
+
+    bene_distinct_cnt_elapsed = round(datetime.utcnow().timestamp() - start_time, 3)
+
+    # Archived grant real/synth bene distinct counts (excludes granted to multiple apps and multiple archived records)
+    start_time = datetime.utcnow().timestamp()
+
+    real_bene_arch_distinct_cnt = (
+        ArchivedDataAccessGrant.objects.filter(
+            ~Q(beneficiary__crosswalk___fhir_id__startswith="-")
+            & ~Q(beneficiary__crosswalk___fhir_id="")
+        )
+        .distinct()
+        .count()
+    )
+
+    synth_bene_arch_distinct_cnt = (
+        ArchivedDataAccessGrant.objects.filter(
+            Q(beneficiary__crosswalk___fhir_id__startswith="-")
+            & ~Q(beneficiary__crosswalk___fhir_id="")
+        )
+        .distinct()
+        .count()
+    )
+
+    bene_arch_distinct_cnt_elapsed = round(
+        datetime.utcnow().timestamp() - start_time, 3
+    )
+
+    # Both Grant and Archived grant (UNION) real/synth bene distinct counts
+    start_time = datetime.utcnow().timestamp()
+
+    real_bene_union_cnt = (
+        DataAccessGrant.objects.filter(
+            ~Q(beneficiary__crosswalk___fhir_id__startswith="-")
+            & ~Q(beneficiary__crosswalk___fhir_id="")
+        )
+        .values("beneficiary__username")
+        .union(
+            ArchivedDataAccessGrant.objects.filter(
+                ~Q(beneficiary__crosswalk___fhir_id__startswith="-")
+                & ~Q(beneficiary__crosswalk___fhir_id="")
+            ).values("beneficiary__username")
+        )
+        .count()
+    )
+
+    synth_bene_union_cnt = (
+        DataAccessGrant.objects.filter(
+            Q(beneficiary__crosswalk___fhir_id__startswith="-")
+            & ~Q(beneficiary__crosswalk___fhir_id="")
+        )
+        .values("beneficiary__username")
+        .union(
+            ArchivedDataAccessGrant.objects.filter(
+                Q(beneficiary__crosswalk___fhir_id__startswith="-")
+                & ~Q(beneficiary__crosswalk___fhir_id="")
+            ).values("beneficiary__username")
+        )
+        .count()
+    )
+
+    bene_union_cnt_elapsed = round(datetime.utcnow().timestamp() - start_time, 3)
+
+    return {
+        "real_bene_cnt": real_bene_cnt,
+        "synth_bene_cnt": synth_bene_cnt,
+        "bene_cnt_elapsed": bene_cnt_elapsed,
+        "real_bene_distinct_cnt": real_bene_distinct_cnt,
+        "synth_bene_distinct_cnt": synth_bene_distinct_cnt,
+        "bene_distinct_cnt_elapsed": bene_distinct_cnt_elapsed,
+        "real_bene_arch_distinct_cnt": real_bene_arch_distinct_cnt,
+        "synth_bene_arch_distinct_cnt": synth_bene_arch_distinct_cnt,
+        "bene_arch_distinct_cnt_elapsed": bene_arch_distinct_cnt_elapsed,
+        "real_bene_union_cnt": real_bene_union_cnt,
+        "synth_bene_union_cnt": synth_bene_union_cnt,
+        "bene_union_cnt_elapsed": bene_union_cnt_elapsed,
+    }
+
+
+def get_grant_by_app_counts(application):
```

Comment:
Maybe we could change these function names to be a bit more descriptive of what data we are trying to get from performing these operations?